syntax = "proto3";
package alert_rule;




import "monitor_sdk/model/monitor/alert_rule.proto";

/*
ListAlertRuleAll请求
*/
message ListAlertRuleAllRequest {
    /*
    如果指定此参数为monitor_v2,则查询的为新告警策略,否则为旧告警策略
    */
    string system = 1 ;
    /*
    sdk限制字段使用
    */
    repeated string XXX_RestFieldMask = 2 ;
}

/*
ListAlertRuleAll返回
*/
message ListAlertRuleAllResponse {
    /*
    code
    */
    int32 code = 1 ;
    /*
    msg
    */
    string msg = 2 ;
    /*
    总数
    */
    int32 total = 3 ;
    /*
    页数
    */
    int32 page = 4 ;
    /*
    分页大小
    */
    int32 page_size = 5 ;
    /*
    data
    */
    repeated monitor.AlertRule data = 6 ;
}

/*
ListAlertRuleAllApi返回
*/
message ListAlertRuleAllResponseWrapper {
    /*
    返回码
    */
    int32 code = 1 ;
    /*
    返回码解释
    */
    string codeExplain = 2 ;
    /*
    错误详情
    */
    string error = 3 ;
    /*
    返回数据
    */
    alert_rule.ListAlertRuleAllResponse data = 4 ;
}

